bindings_ffi/src/lib.rs

```diff
@@ -69,41 +69,41 @@ pub async fn create_client(
     init_logger(logger);
 
     let inbox_owner = RustInboxOwner::new(ffi_inbox_owner);
-    let api_client = TonicApiClient::create(host.clone(), is_secure)
-        .await
-        .map_err(|e| stringify_error_chain(&e))?;
+    log::info!(
+        "Creating API client for host: {}, isSecure: {}",
+        host,
+        is_secure
+    );
+    let api_client = TonicApiClient::create(host.clone(), is_secure).await?;
 
+    log::info!(
+        "Creating message store with path: {:?} and encryption key: {}",
+        db,
+        encryption_key.is_some()
+    );
     let key: EncryptionKey = match encryption_key {
-        Some(key) => key.try_into().unwrap(),
+        Some(key) => key.try_into().map_err(|_err| GenericError::Generic {
+            err: "Malformed 32 byte encryption key".to_string(),
+        })?,
         None => static_enc_key(),
     };
 
     let store = match db {
-        Some(path) => {
-            info!("Using persistent storage: {} ", path);
-            EncryptedMessageStore::new(StorageOption::Persistent(path), key)
-        }
+        // TODO support encryption
+        Some(path) => EncryptedMessageStore::new_unencrypted(StorageOption::Persistent(path)),
```

Comment:
Disable DB encryption for now, seeing as it doesn't work correctly. Doesn't affect the bindings interface